File: demos/dice-game/typescript/dice-game.ts
function rollDice(): number {
    return Math.floor(Math.random() * 6) + 1;
}

function playGame(): number {
    let totalScore = 0;
    for (let i = 0; i < 10; i++) {
        const dice1 = rollDice();
        const dice2 = rollDice();
        let score = dice1 + dice2;
        if (dice1 === dice2) {
            score *= 2;
        }
        totalScore += score;
    }
    return totalScore;
}

function main(): void {
    const player1Score = playGame();
    const player2Score = playGame();

    if (player1Score > player2Score) {
        console.log(`Player 1 wins with ${player1Score} points`);
    } else if (player2Score > player1Score) {
        console.log(`Player 2 wins with ${player2Score} points`);
    } else {
        console.log(`It's a tie with ${player1Score} points each`);
    }
}

main();

File: demos/dice-game/golang/dice-game.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func rollDice() int {
	return rand.Intn(6) + 1
}

func playGame() int {
	totalScore := 0
	for i := 0; i < 10; i++ {
		dice1 := rollDice()
		dice2 := rollDice()
		score := dice1 + dice2
		if dice1 == dice2 {
			score *= 2
		}
		totalScore += score
	}
	return totalScore
}

func main() {
	rand.Seed(time.Now().UnixNano())

	player1Score := playGame()
	player2Score := playGame()

	if player1Score > player2Score {
		fmt.Printf("Player 1 wins with %d points\n", player1Score)
	} else if player2Score > player1Score {
		fmt.Printf("Player 2 wins with %d points\n", player2Score)
	} else {
		fmt.Printf("It's a tie with %d points each\n", player1Score)
	}
}
